### PR TITLE
add refactoring for go-mode

### PR DESCRIPTION
**Pull request by TinySong**
~SPC m r~ for gorename move to ~SPC m R~
~SPC m r~ for refactoring

### CODE REVIEW

**Comment by LemmingAvalanche** [comment]
It looks like you used Org code syntax in your commit message, which GitHub
MarkDown then interprets as strikethrough in your description. As a result it
looks like you changed your mind on this pull request. 😛

I think using MarkDown-ish syntax in commit messages is more conventional than
using Org syntax.

File: layers/+lang/go/packages.el [hunk 1]
```diff
@@ -67,25 +68,25 @@
                                    "-check.f"
                                  "-run")))
               (save-excursion
-                  (re-search-backward "^func[ ]+\\(([[:alnum:]]*?[ ]?[*]?[[:alnum:]]+)[ ]+\\)?\\(Test[[:alnum:]_]+\\)(.*)")
```
Comment (LemmingAvalanche) [inline layers/+lang/go/packages.el]: The change on this hunk is non-functional and probably accidental; it just de-indents the code by two spaces. This makes it look like you changed more code than you “really” did, which can be distracting.

File: layers/+lang/go/packages.el [hunk 2]
```diff
@@ -67,25 +68,25 @@
...
-                  (spacemacs/go-run-tests (concat test-method "='" (match-string-no-properties 2) "'"))))
```
Comment (LemmingAvalanche) [inline layers/+lang/go/packages.el]: Same as above hunk (de-indents by two spaces).

File: layers/+lang/go/packages.el [hunk 3]
```diff
@@ -67,25 +68,25 @@
...
+                (re-search-backward "^func[ ]+\\(([[:alnum:]]*?[ ]?[*]?[[:alnum:]]+)[ ]+\\)?\\(Test[[:alnum:]_]+\\)(.*)")
+                (spacemacs/go-run-tests (concat test-method "='" (match-string-no-properties 2) "'"))))
           (message "Must be in a _test.go file to run go-run-test-current-function")))
 
       (defun spacemacs/go-run-test-current-suite ()
         (interactive)
         (if (string-match "_test\.go" buffer-file-name)
             (if go-use-gocheck-for-testing
                 (save-excursion
-                    (re-search-backward "^func[ ]+\\(([[:alnum:]]*?[ ]?[*]?\\([[:alnum:]]+\\))[ ]+\\)?Test[[:alnum:]_]+(.*)")
```
Comment (LemmingAvalanche) [inline layers/+lang/go/packages.el]: Same as above hunk (de-indents by two spaces).

File: layers/+lang/go/packages.el [hunk 4]
```diff
@@ -67,25 +68,25 @@
...
-                    (spacemacs/go-run-tests (concat "-check.f='" (match-string-no-properties 2) "'")))
```
Comment (LemmingAvalanche) [inline layers/+lang/go/packages.el]: Same as above hunk (de-indents by two spaces).

File: layers/+lang/go/packages.el [hunk 5]
```diff
@@ -67,25 +68,25 @@
...
+                  (re-search-backward "^func[ ]+\\(([[:alnum:]]*?[ ]?[*]?\\([[:alnum:]]+\\))[ ]+\\)?Test[[:alnum:]_]+(.*)")
+                  (spacemacs/go-run-tests (concat "-check.f='" (match-string-no-properties 2) "'")))
               (message "Gocheck is needed to test the current suite"))
           (message "Must be in a _test.go file to run go-test-current-suite")))
 
       (defun spacemacs/go-run-main ()
         (interactive)
         (shell-command
-          (format "go run %s"
```
Comment (LemmingAvalanche) [inline layers/+lang/go/packages.el]: Same as above hunk (de-indents by two spaces).

File: layers/+lang/go/packages.el [hunk 6]
```diff
@@ -67,25 +68,25 @@
...
-                  (shell-quote-argument (buffer-file-name)))))
```
Comment (LemmingAvalanche) [inline layers/+lang/go/packages.el]: Same as above hunk (de-indents by two spaces).

**Comment by TinySong** [comment]
@LemmingAvalanche  Very Thanks your advice

**Pull request closed without merging**